### PR TITLE
Refactored table component, clearInterval on unmount, updated sort logic without state mutations

### PR DESCRIPTION
**Pull request by crup**
# Cleaned table.js
- Removed functions calls from table.js `doSort` and `setInterval`. These function calls should be effects. 
- Fixes existing sort which was mutating the component state instead of setting new one.
- Fixes an effect where `setCount` was being called recursively.
- Cleaned `onHighlightState` method.
- Refactored table rows to array map.

### CODE REVIEW

**Comment by jeremyphilemon** [comment]
This looks good! Could you rebase please? :)

**Comment by crup** [comment]
Updated with master.

**Comment by crup** [comment]
Reverted my sort implementation, looks like I forgot to handle `state`'s case in previous PR. So, for my I'm sticking with existing sort logic.

File: src/components/table.js [hunk 1]
```diff
@@ -16,30 +16,28 @@ function Table(props) {
...
+      const affectedStates = props.states.reduce((pv, cv, ci) => {
+        return cv.confirmed > 0 && ci > 0 ? pv + 1 : pv;
+      }, 0);
+      setCount(affectedStates);
     }
-  }, [states.length]);
+  }, [props.states.length]);
 
   useEffect(()=>{
     getDistricts();
```
Comment (vramana) [inline src/components/table.js:43]: Can you also remove the 1 in the inputs array here? It does not matter at all.

File: src/components/table.js [hunk 2]
```diff
@@ -56,113 +54,101 @@ function Table(props) {
...
+    const interval = setInterval(getDistricts, 600000);
+    return () => clearInterval(interval);
+  }, []);
+
+  const sortByKey = (arr, key, isAscending) => {
+    return arr.concat().sort((s1, s2) => {
+      const v1 = key !== 'state' ? parseInt(s1[key]) : s1[key];
+      const v2 = key !== 'state' ? parseInt(s2[key]) : s2[key];
+      if (isAscending) {
+        return v1 > v2 ? 1 : (v1 == v1) && s1['state'] > s2['state'] ? 1 : -1;
```
Comment (vramana) [inline src/components/table.js]: Why not multiply this value by -1 if isAscending is false?  That will remove the else branch.

File: src/components/table.js [hunk 3]
```diff
@@ -56,113 +54,101 @@ function Table(props) {
...
+        return v1 < v2 ? 1 : (v1 == v2) && s1['state'] > s2['state'] ? 1 : -1;
       }
     });
-    {/* console.log(states);*/}
-    states.unshift(totalRow[0]);
   };
 
-  const handleSort = (e, props) => {
+  useEffect(() => {
+    const data = [...props.states].slice(1, props.states.length);
```
Comment (vramana) [inline src/components/table.js]:
```js
const data = props.states.slice(1) 
```
This has the same effect.

File: src/components/table.js (hunk 2 above)
Comment (crup) [reply src/components/table.js]: Cool! Fixed it.

File: src/components/table.js (hunk 1 above)
Comment (crup) [reply src/components/table.js:43]: Fixed it. It was a miss.

File: src/components/table.js (hunk 3 above)
Comment (crup) [reply src/components/table.js]: Done!

File: src/components/table.js [hunk 4]
```diff
@@ -56,117 +54,100 @@ function Table(props) {
...
     });
-    {/* console.log(states);*/}
-    states.unshift(totalRow[0]);
   };
 
-  const handleSort = (e, props) => {
-    const currentsortColumn = e.currentTarget.querySelector('abbr').getAttribute('title').toLowerCase();
-    const isAscending = sortData.sortColumn == currentsortColumn? !sortData.isAscending : sortData.sortColumn === 'state';
+  useEffect(() => {
+    const data = props.states.slice(1, props.states.length);
```
Comment (vramana) [inline src/components/table.js]: You don't need to mention props.state.length. It automatically slices till the end if you omit it. 

```
props.states.slice(1)
```
This is good enough

Comment (crup) [reply src/components/table.js]: Done!!

**Comment by jeremyphilemon** [comment]
Could you please squash and rebase your commits? :)

**Comment by crup** [comment]
Too much pain to squash and resolve conflicts. Will create a new branch from current master and do the changes.

**Comment by jeremyphilemon** [comment]
@crup I understand, apologies for the backlog and the eventual conflicts! Will keep an eye out for your PR! 

Also create an issue and have yourself assigned so we can keep track of it better!

**Pull request closed without merging**